# Final Model Evaluation


The original 100-epoch `best.pt` model was selected as the final model because additional fine-tuning and a larger image size did not provide a meaningful improvement.

This notebook evaluates the final model on the test dataset

In [3]:
# Import YOLO
from ultralytics import YOLO


In [4]:
# Load the best model from the 100-epoch training run

model = YOLO("runs/detect/helmet_100_epochs/weights/best.pt")

## Test Set Evaluation

The final model is now evaluated on the test dataset.


In [5]:
# Evaluate the final model on the test images

test_results = model.val(
    data="../data/data.yaml",
    split="test",
    imgsz=640,
    batch=2,
    device=0,
    name="final_test"
)

Ultralytics 8.4.135  Python-3.11.16 torch-2.13.0+cu132 CUDA:0 (NVIDIA GeForce GTX 1650 Ti with Max-Q Design, 4096MiB)
YOLO26n summary (fused): 122 layers, 2,375,226 parameters, 0 gradients, 5.3 GFLOPs
WARNING val: Slow image access detected (ping: 0.30.2 ms, read: 16.73.9 MB/s, size: 101.4 KB). Use local storage instead of remote/mounted storage for better performance. See https://docs.ultralytics.com/guides/model-training-tips
val: Scanning D:\ML\realtime-helmet-detection\data\test\labels... 184 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 184/184 519.7it/s 0.4s0.1s
val: New cache created: D:\ML\realtime-helmet-detection\data\test\labels.cache
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 92/92 29.1it/s 3.2s0.1s
                   all        184        434      0.814       0.71      0.802      0.458
                Helmet        122        233      0.841      0.729      0.824      0.487
             No-Helmet        

## Final Test Results

The final model was evaluated on the test dataset, which was not used during model tuning.

The final test results were:

- Precision: 0.814
- Recall: 0.710
- mAP@50: 0.802
- mAP@50-95: 0.458

The test results were slightly lower than the validation results, which is expected because the test set contains unseen data.

The model performed better on the Helmet class than the No-Helmet class, but both classes achieved reasonable detection performance.

These results will be kept as the final evaluation of the detection model.